modules/library/main/src/test/java/org/geotools/geometry/jts/ReferencedEnvelopeTest.java

```diff
@@ -12,16 +12,31 @@
 import java.awt.geom.Point2D;
 import java.awt.geom.Rectangle2D;
 import org.geotools.api.geometry.MismatchedReferenceSystemException;
+import org.geotools.api.referencing.FactoryException;
 import org.geotools.api.referencing.crs.CoordinateReferenceSystem;
 import org.geotools.geometry.GeneralBounds;
 import org.geotools.referencing.CRS;
 import org.geotools.referencing.crs.DefaultEngineeringCRS;
 import org.geotools.referencing.crs.DefaultGeographicCRS;
+import org.geotools.referencing.crs.DefaultProjectedCRS;
+import org.geotools.util.factory.GeoTools;
+import org.geotools.util.factory.Hints;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 import org.locationtech.jts.geom.Coordinate;
 
 public class ReferencedEnvelopeTest {
 
+    @Before
+    public void setUp() throws Exception {
+        // this is the only thing that actually forces CRS object to give up
+        // its configuration, necessary when tests are run by Maven, one JVM for all
+        // the tests in this module
+        Hints.putSystemDefault(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER, Boolean.FALSE);
+        GeoTools.fireConfigurationChanged();
```

Comment:
Mind a "multithreaded" build is really a "multiprocess" one, surefire uses a pool of JVMs, one per module, that are re-used across modules. The tests for a single module are executed sequentially by a single JVM. So there is no need to protect against classic threading problems, but one needs to be sure that the baseline configuration is restored before the test exits (e.g., @BeforeClass and @AfterClas), to avoid surprises in the test class or module in the build.